### ISSUE TITLE
Replace i18n.t() with tpl() everywhere

### ISSUE THREAD

**Issue by ErisDS** [TryGhost/Ghost#13380]
Our internal, server-side i18n package is deprecated, and so is the use of `i18n.t()`. 

Instead we have a new package [@tryghost/tpl](https://www.npmjs.com/package/@tryghost/tpl) that does string interpolation and makes refactoring i18n out of the codebase easier.   

We have successfully removed `i18n.t()` from core/frontend, but the rest of the codebase has many many references that need to be updated. At the time of writing there are 404 instances of i18n.t in 143 files!

## Steps to refactor

1. Wherever there's an instance of i18n.t in a file we need to:

Replace  
`const i18n.t = require('../[some nesting]../core/shared/i18n)`

With

`const tpl = require('@tryghost/tpl');`

2. Then at the top of the document, after the last require, add a new messages object:

```
const messages = {
};
```

That's the setup done for this file.

3. For each instance of `i18n.t()` for example:

`i18n.t('common.warnings.helpers.flagMustBeEnabled', {flagName: 'something'});`

You'll need to go into `core/shared/i18n/translations/en.json` and find the correct string.

 E.g 
`common.warnings.helpers.flagMustBeEnabled` 

Will look like:
`"flagMustBeEnabled": "The {flagName} flag must be enabled in labs if you wish to use the \\{\\{{helperName}\\}\\} helper.",`

Copy this string into your messages object, as is. If you have your editor setup to use our eslint rules & set to auto fix or auto fix on save, it should automatically fix the quotes & commas.

⚠️ Don't cut or otherwise remove the string from en.json. Strings are reused in multiple places, and it's a waste of time trying to figure out if a string is now unused. We'll simply delete the file verbatim once this refactor is finished.

You should now have something like this:

```
const tpl = require('@tryghost/tpl');
// more requires

const messages = {
	flagMustBeEnabled: 'The {flagName} flag must be enabled in labs if you wish to use the \\{\\{{helperName}\\}\\} helper.'
};
```

Finally, you need to change the reference to `i18n.t` to `tpl` and the string path, to the messages variable:

So from:

`i18n.t('common.warnings.helpers.flagMustBeEnabled', {flagName: 'something'});`

To:

`tpl(messages.flagMustBeEnabled, {flagName: 'something'});`

And that's it!

There is no need to mess with the escaping or interpolation!

## How to work on this issue
The slow and stead approach to refactoring is the only thing that realistically works. Branches should be very short lived (<1 day if possible). Therefore, please do not attempt to do this entire refactor in a single PR. It is not possible for us to review!

Instead, please find a single file with many references, or a small subfolder of the codebase with a handful of usages, make the changes, and submit a PR. Then repeat.

Please follow our contribution guidelines as closely as you can, particularly being sure to reference this issue in your commit. Refactor commits should not use emojis as they are not user-facing changes.

Note: This refactor has almost no user facing impact and no tests should break or change. The en.json file should also not change as a result of this work.

Thank you 🙏 ❤️  🙏 

---

**Why is `i18n.t` deprecated?**

💡  Removing this does not at all impact the current or future ability to translate Ghost. 

This was a concept introduced years ago with a view to aiding translating Ghost. Most people are looking to either translate their theme (not affected by this refactor) or have a translated Ghost Admin panel, which is not solved by this utility. Most people are not interested in translating errors from the theme or JSON APIs, which is effectively what this util would do if we allowed alternative translation files. Meanwhile, this utility creates tight coupling throughout the Ghost codebase which is slowing us down and preventing us from shipping things we do want.

**Comment by neverett8fr** [comment]
Hey!   
I had a look at the code - In the issue you mentioned the `i18n` package is deprecated and to replace the `i18n.t` files.  
Out of interest does that mean the `i18n.test.js` unit test file should be removed?  
  
Potentially a new issue, for after this one's closed?  
  
![image](https://user-images.githubusercontent.com/43852724/135655233-2c33396c-ad53-43bd-936b-d5ecbe533157.png)

**Comment by ErisDS** [comment]
@neverett8fr no - removing the test file would not be a valid change as the library is used as the basis for theme i18n.

Thank you to everyone who has contributed towards this effort so far! It's greatly appreciated.

I would love it if we could prioritise getting the `core/server/web` folder cleaned up first. Thank you 🙏

**Comment by ErisDS** [comment]
Thank you to everyone who has submitted a PR so far. I've officially added the hacktoberfest topic & labels so all PRs should automatically get accepted once merged.

I've also just updated the issue to provide more details about how to check your work. Many PRs are failing the build atm with minor linting issues.

**Comment by aniketh-varma** [comment]
Hi, is this issue still open? If it is I can help with a few files.

**Comment by ErisDS** [comment]
Thank you again to everyone who has helped out so far. We are getting there 🙏 

There are still over 300 references to update though! 😅 

Please keep the PRs coming!
